lib/deref.js

```diff
@@ -383,5 +389,67 @@ module.exports = {
     }
 
     return schema;
+  },
+
+  /**
+   * Take a $ref and a spec and return the referenced value
+   * @param {object} spec the parsed spec object
+   * @param {string} reference the $ref value to dereference
+   * @returns {object} the dereferenced $ref value
+   */
+  dereferenceElement: function (spec, reference) {
+    let splitRef = reference.split('/'),
+      resolvedContent;
+
+    splitRef = splitRef.slice(1).map((elem) => {
+      // https://swagger.io/docs/specification/using-ref#escape
+      // since / is the default delimiter, slashes are escaped with ~1
+      return decodeURIComponent(
```

Comment:
We may extract this to a function with a meaningful name and get rid of the comment, we could add the comment in the function docs